### DISCUSSION TITLE
Is it possible to create a box container widget with this?

### DISCUSSION THREAD

**Discussion by martinRenou** [manzt/anywidget#580] [General]
We tried creating a box container with this, a box that contains other widgets. It seems it's not possible by design?

```python
import pathlib
import anywidget
import traitlets

class Box(anywidget.AnyWidget):
    _esm = pathlib.Path("box.js")
    _css = pathlib.Path("box.css")
    children = traitlets.List(traitlets.Instance(anywidget.AnyWidget)).tag(sync=True)
```

Using plain old ipywidgets approach, the frontend Box view code would call the `create_child_view` method to create sub widgets. But there is no such thing available from the anywidgets API as widget authors don't have access to the view implementation, or am I missing something?

**Comment by manzt** [comment]
> It seems it's not possible by design?

Correct. Currently we've avoided adding additional APIs for creating nested views and child views. It is possible to use anywidgets within ipywidgets layouts as well as Panel layout containers by making this restriction.

 I'd be open to brainstorming some APIs to suit such use cases, but I'd like to avoid tying the APIs so closely to backbone model/view-based approaches.

**Comment by maartenbreddels** [comment]
For ipyreact this was a bit more trival to implement, because React supports this out of the box.

Maybe can be an inspiration for anywidget how to handle that, I have a demo for this at:
https://py.cafe/maartenbreddels/ipyreact-button-layout

We spend quite some time to get this right, for non-dom react children as well, e.g. threejs-fiber:
https://py.cafe/maartenbreddels/solara-ipyreact-threejs-fiber-boxes
